### ISSUE TITLE
Managing secondary threads

### ISSUE THREAD

**Issue by vadosnaprimer** [clementgallet/libTAS#288]
I remember I was told that you can't automatically detect when a secondary thread spawns that we need to wait for before resuming the movie. Those cases when you need to pause libTAS and explicitly wait for some time (Celeste, SteamWorld Dig 2).

Even if those can't be automatically detected, you still find out they've appeared and you need to pause on that frame, right? I noticed it needs different amounts of waiting time depending on the machine speed. Would it be possible to explicitly record in the movie "wait for this thread to finish before advancing the frame"? I think it's called "joining" a thread. That would hopefully automate the pauses upon replaying, and hopefully account of machine specs in terms of delay time.

**Comment by clementgallet** [comment]
There was a feature like that implemented years ago in libTAS, which I had to removed when savestating was implemented (because I needed to rewrite thread handling from scratch). At that time, after some testing with several games, I noticed that very few games use threads for loading with that pattern (spawning a thread when loading begins, and joining/detaching the thread when it ends). Also, those threads need to be identified, so we need at least some mechanism (either manual toggling, or some auto-detection). So at that time, I decided that it was not worth re-implementing this in libTAS.

Now, I guess I could implement something again. It definitively won't help for SWD2, as the loading thread is running the entire time, and synchronization is done through variables (not even atomic, so undetectable by libTAS).
